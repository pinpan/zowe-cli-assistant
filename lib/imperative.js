"use strict";
/**
 * This program and the accompanying materials are made available and may be used, at your option, under either:
 * * Eclipse Public License v2.0, available at https://www.eclipse.org/legal/epl-v20.html, OR
 * * Apache License, version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
 *
 * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
 *
 * Copyright Contributors to the Zowe Project.
 */
//import { ProfileConstants } from "./api/ProfileConstants";
const config = {
    commandModuleGlobs: ["**/cli/*/*.definition!(.d).*s"],
    pluginHealthCheck: __dirname + "/healthCheck.Handler",
    pluginSummary: "Zowe CLI command typing assistant",
    pluginAliases: ["zcta"],
    rootCommandDescription: "Welcome to the Assistant plug-in for Zowe CLI!\n\n This plug-in " +
        "(& CLI) follows the Zowe CLI command syntax 'zowe [group] [action] [object] [options]'. " +
        "Where, in the case of the plugin, " +
        "the [group] is the package.json name, " +
        "the [actions] are defined in the project under 'src/cli/', " +
        "& the [objects] (definitions & handler) are defined in the project under 'src/cli/'.",
    productDisplayName: "Zowe CLI Assistant",
    name: "zowe-cli-assistant",
};
module.exports = config;
//# sourceMappingURL=imperative.js.map