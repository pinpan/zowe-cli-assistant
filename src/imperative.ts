/**
 * This program and the accompanying materials are made available and may be used, at your option, under either:
 * * Eclipse Public License v2.0, available at https://www.eclipse.org/legal/epl-v20.html, OR
 * * Apache License, version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
 *
 * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
 *
 * Copyright Contributors to the Zowe Project.
 */

import {IImperativeConfig} from "@zowe/imperative";
//import { ProfileConstants } from "./api/ProfileConstants";

const config: IImperativeConfig = {
    commandModuleGlobs: ["**/cli/*/*.definition!(.d).*s"],
    pluginHealthCheck: __dirname + "/healthCheck.Handler",
    pluginSummary: "Zowe CLI command typing assistant",
    pluginAliases: ["zcta"],
    rootCommandDescription: "Welcome to the Assistant plug-in for Zowe CLI!\n\n This plug-in " +
        "(& CLI) follows the Zowe CLI command syntax 'zowe [group] [action] [object] [options]'. " +
        "Where, in the case of the plugin, " +
        "the [group] is the package.json name, " +
        "the [actions] are defined in the project under 'src/cli/', " +
        "& the [objects] (definitions & handler) are defined in the project under 'src/cli/'.",
    productDisplayName: "Zowe CLI Assistant",
    name: "zowe-cli-assistant",
    /*profiles: [
        {
            type: "sample",
            schema: {
                type: "object",
                title: "Sample Profile",
                description: "A sample profile is required to issue commands that interact with the sample API. The " +
                    "sample profile contains your host, port, user name, and password to connect to the sample API.",
                properties: {
                    host: {
                        type: "string",
                        optionDefinition: ProfileConstants.SAMPLE_OPTION_HOST
                    },
                    port: {
                        type: "number",
                        includeInTemplate: true,
                        optionDefinition: ProfileConstants.SAMPLE_OPTION_PORT
                    },
                    user: {
                        type: "string",
                        secure: true,
                        optionDefinition: ProfileConstants.SAMPLE_OPTION_USER
                    },
                    password: {
                        type: "string",
                        secure: true,
                        optionDefinition: ProfileConstants.SAMPLE_OPTION_PASSWORD
                    },
                    rejectUnauthorized: {
                        type: "boolean",
                        optionDefinition: ProfileConstants.SAMPLE_OPTION_REJECT_UNAUTHORIZED
                    },
                    tshirtSize: {
                        type: "string",
                        optionDefinition: ProfileConstants.SAMPLE_OPTION_TSHIRT_SIZE
                    }
                }
            }
        }
    ]*/
};

export = config;
