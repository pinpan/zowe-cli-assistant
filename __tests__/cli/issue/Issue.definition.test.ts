/**
 * This program and the accompanying materials are made available and may be used, at your option, under either:
 * * Eclipse Public License v2.0, available at https://www.eclipse.org/legal/epl-v20.html, OR
 * * Apache License, version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
 *
 * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
 *
 * Copyright Contributors to the Zowe Project.
 */

import { ICommandDefinition } from "@zowe/imperative";
import { relative, sep, posix } from "path";

describe("Issue definition", () => {
    it("should match the snapshot", () => {
        const contents = require("../../../src/cli/issue/Issue.definition");
        contents.children.forEach((child: ICommandDefinition) => child.handler = relative(__dirname, child.handler).split(sep).join(posix.sep));
        expect(contents).toMatchSnapshot();
    });
});
