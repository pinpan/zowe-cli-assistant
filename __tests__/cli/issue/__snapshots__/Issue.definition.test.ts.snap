// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Issue definition should match the snapshot 1`] = `
Object {
  "children": Array [
    Object {
      "aliases": Array [
        "cwo",
      ],
      "description": "[objects] in Zowe CLI are the entities on which [actions] are perfomed. [objects] are always nouns. For example, for command \\"zowe zos-files delete data-set\\", the [object] is \\"data-set\\".

For this command, we require options to be specified.",
      "handler": "../../../src/cli/issue/command_with_options/CommandWithOptions.handler",
      "name": "command-with-options",
      "options": Array [
        Object {
          "description": "Boolean flag, that if specified, requires that you also specify a string for option \\"the string\\"",
          "implies": Array [
            "the-string",
          ],
          "name": "use-the-string",
          "type": "boolean",
        },
        Object {
          "description": "A string type option that is required if \\"use-the-string\\" is specified.",
          "name": "the-string",
          "type": "string",
        },
        Object {
          "description": "A required number.",
          "name": "required-number",
          "required": true,
          "type": "number",
        },
      ],
      "summary": "[object] command-with-options requires options",
      "type": "command",
    },
    Object {
      "aliases": Array [
        "cwp",
      ],
      "description": "[objects] in Zowe CLI are the entities on which [actions] are perfomed. [objects] are always nouns. For example, for command \\"zowe zos-files delete data-set\\", the [object] is \\"data-set\\". For this command, we will simply print console error (stderr) messages and the command should succeed.",
      "handler": "../../../src/cli/issue/command_with_positionals/CommandWithPositionals.handler",
      "name": "command-with-positionals",
      "positionals": Array [
        Object {
          "description": "The first positional parameter",
          "name": "firstpositional",
          "required": true,
          "type": "string",
        },
        Object {
          "description": "The second positional parameter",
          "name": "secondpositional",
          "required": true,
          "type": "number",
        },
      ],
      "summary": "[object] command-with-positionals prints console msgs",
      "type": "command",
    },
  ],
  "description": "[actions] in Zowe are groups of commands. [actions] are always verbs. For example, for command \\"zowe zos-files delete data-set\\", the [action] is \\"delete\\".

For this action (\\"issue\\") we will issue commands with required/optional/positional options.",
  "name": "issue",
  "summary": "Perform \\"the issue [action]\\" on some [objects]",
  "type": "group",
}
`;
